docs/index.rst

```diff
@@ -1,63 +1,176 @@
 ``pluggy``
 ==========
+**The pytest plugin system**
 
-The ``pytest`` plugin system
-****************************
+What is it?
+***********
 ``pluggy`` is the crystallized core of `plugin management and hook
-calling`_ for `pytest`_.
+calling`_ for `pytest`_. It enables `200+ plugins`_ to extend and customize
+``pytest``'s default behaviour. Even ``pytest`` itself is composed
+as a set of ``pluggy`` plugins which are invoked in sequence according to a
+well defined set of protocols.
+
+It gives users the ability to extend or modify the behaviour of a
+``host program`` by installing a ``plugin`` for that program.
+The plugin code will run as part of normal program execution, changing or
+enhancing certain aspects of it.
+
+In essence, ``pluggy`` enables function `hooking`_ so you can build
+"pluggable" systems.
+
+Why is it useful?
+*****************
+There are some established mechanisms for modifying the behavior of other
+programs/libraries in Python like
+`method overriding <https://en.wikipedia.org/wiki/Method_overriding>`_
+(e.g. Jinja2) or
+`monkey patching <https://en.wikipedia.org/wiki/Monkey_patch>`_ (e.g. gevent
+or for `writing tests <https://docs.pytest.org/en/latest/monkeypatch.html>`_).
+These strategies become problematic though when several parties want to
+participate in the modification of the same program. Therefore ``pluggy``
+does not rely on these mechanisms to enable a more structured approach and
+avoid unnecessary exposure of state and behaviour. This leads to a more
+`loosely coupled <https://en.wikipedia.org/wiki/Loose_coupling>`_ relationship
+between ``host`` and ``plugins``.
+
+The ``pluggy`` approach puts the burden on the designer of the
+``host program`` to think carefully about which objects are explicitly needed
+by an implementation and gives designers of the ``plugin`` a clear framework
+for how to extend the ``host`` by giving them a well defined set of functions
+and objects to work with.
+
+How does it work?
+*****************
+Let us start with a short overview of what is involved:
+
+* ``host`` or ``host program``: the program offering extensibility
+  by specifying ``hook functions`` and invoking their implementation(s) as
+  part of program execution
+* ``plugin``: the program implementing a subset of the specified hooks and
```

Comment:
`a subset of` should maybe be inside `()` since technically a plugin could implement all hooks.
Though I guess the a set is a subset of itself?

Reply:
I had the same thoughts myself and came to the latter conclusion, but if you had to do the same mental gymnastics, I guess it is better to add the parens :)